compositor_pipeline/src/pipeline/pipeline_input.rs

```diff
@@ -56,7 +65,10 @@ pub(super) fn register_pipeline_input<NewInputResult>(
         }
     }
 
-    guard.inputs.insert(input_id.clone(), pipeline_input);
+    match guard.inputs.entry(input_id.clone()) {
```

Comment:
if you do that, you are leaving partially modified state, you need to abort it before output stream end condition is modified